samza-core/src/main/java/org/apache/samza/zk/ZkLeaderElector.java

```diff
@@ -105,7 +91,7 @@ public boolean tryBecomeLeader() {
     if (index == 0) {
       isLeader.getAndSet(true);
       LOGGER.info(zLog("Eligible to become the leader!"));
-      zkLeaderElectorListener.onBecomingLeader(); // inform the caller
+      leaderElectorListener.onBecomingLeader(); // inform the caller
```

Comment:
> if you create a LeaderElector object in one place and pass it to another class to use.

I am not following the scenario you are suggesting. Can you please give an example?